Layers/Renderers/Terrain/TerrainLOD.gd

```diff
@@ -94,47 +94,48 @@ func build():
 			top_left_x,
 			top_left_y,
 			size,
-			mesh_resolution,
-			1
+			landuse_resolution,
+			0
 		)
 		
-		if current_surface_height_image.is_valid():
-			current_surface_heightmap = current_surface_height_image.get_image_texture()
-	
-	if landuse_layer and (always_load_landuse or load_detail_textures or load_fade_textures):
-		if load_detail_textures or load_fade_textures:
-			var most_common_groups = current_landuse_image.get_most_common(MAX_GROUPS)
-			var group_array = Vegetation.get_group_array_for_ids(most_common_groups)
-			
-			current_metadata_map = Vegetation.get_metadata_map(most_common_groups)
-			
-			if load_detail_textures:
-				current_albedo_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "albedo")
-				current_normal_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "normal")
-				current_specular_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "specular")
-				current_ambient_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "ambient")
-				current_roughness_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "roughness")
+		if current_landuse_image.is_valid():
+			current_landuse = current_landuse_image.get_image_texture()
+		
+		# Surface Height
+		if surface_height_layer:
+			var current_surface_height_image = surface_height_layer.get_image(
+				top_left_x,
+				top_left_y,
+				size,
+				mesh_resolution,
+				1
+			)
 			
-			if load_fade_textures:
-				current_albedo_fade_textures = Vegetation.get_fade_sheet_texture(group_array, "albedo")
-				current_normal_fade_textures = Vegetation.get_fade_sheet_texture(group_array, "normal")
+			if current_surface_height_image.is_valid():
+				current_surface_heightmap = current_surface_height_image.get_image_texture()
+		
+		if landuse_layer and (always_load_landuse or load_detail_textures or load_fade_textures):
+			if load_detail_textures or load_fade_textures:
+				var most_common_groups = current_landuse_image.get_most_common(MAX_GROUPS)
+				var group_array = Vegetation.get_group_array_for_ids(most_common_groups)
+				
+				current_metadata_map = Vegetation.get_metadata_map(most_common_groups)
+				
+				if load_detail_textures:
+					current_albedo_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "albedo")
+					current_normal_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "normal")
+					current_specular_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "specular")
+					current_ambient_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "ambient")
+					current_roughness_ground_textures = Vegetation.get_ground_sheet_texture(group_array, "roughness")
+				
+				if load_fade_textures:
+					current_albedo_fade_textures = Vegetation.get_fade_sheet_texture(group_array, "albedo")
+					current_normal_fade_textures = Vegetation.get_fade_sheet_texture(group_array, "normal")
 
 func apply_textures():
 	rebuild_aabb()
```

Comment:
Don't really understand what is going on here, but i guess it is not breaking :D

Reply:
I think that was a fix to get the BasicTerrainRenderer to work again. I'll check that and cherry-pick those changes to main if that's the case